helpers/__tests__/request-test.ts

```diff
@@ -87,6 +136,45 @@ describe('parseText', () => {
   });
 });
 
+describe('postJSON', () => {
+  it('should post without parameters and get json', async () => {
+    const response = new Response('{}');
+    response.json = jest.fn().mockResolvedValue({});
+    window.fetch = jest.fn().mockResolvedValue(response);
+    postJSON(url);
+    await new Promise(setImmediate);
+
+    expect(window.fetch).toBeCalledWith(url, expect.objectContaining({ method: 'POST' }));
+    expect(response.json).toBeCalled();
+  });
+
+  it('should post with a body and get json', () => {
+    postJSON(url, { data: 'test' });
+    expect(window.fetch).toBeCalledWith(
+      url,
+      expect.objectContaining({ body: 'data=test', method: 'POST' })
+    );
+  });
+});
+
+describe('post', () => {
+  it('should post without parameters and return nothing', async () => {
+    const response = new Response('{}');
+    response.json = jest.fn().mockResolvedValue({});
+    response.text = jest.fn().mockResolvedValue('');
```

Comment:
What about a mock function that create a response based on a value, a status code, etc.

```ts
const response = mockResponse(undefined, undefined, {});
function mockResponse(headers: T.Dict<string> = {}, status = 200, value?: any): any {
  return {
    headers: { get: (prop: string) => headers[prop] },
    status,
    json: jest.fn().mockReturnValue(value ? JSON.stringify(value) : undefined),
    text: jest.fn().mockReturnValue(value ? `${value}` : undefined)
  };
}
```
Or something like that. I did not think it all the way through but a lot of code in this file is about creating the response.

Reply:
Good idea!